### ISSUE TITLE
stack ghci doesn't accept ghc-options

### ISSUE THREAD

**Issue by bergmark** [commercialhaskell/stack#971]
Woops, accidental submit \*typing\*

**Comment by bergmark** [comment]
This seems to be a change in 0.1.4.0 (I Just upgraded to the bindist, `Version 0.1.4.0, Git revision dac8cfe51802e08edb523b80c51cf35e21933513 (1722 commits) X86_64`)

Previously my workflow was `stack build --pedantic` and then `stack ghci a-package` if a-package failed to compile. in 0.1.4.0 this happens:

```
$ stack ghci package
tostring-0.2.1.1-0dadd4186e26cc4bb0de459985e35fa9: unregistering (flags changed from ["--ghc-options","-Wall","--ghc-options","-Werror"] to [])
```

`ghc --pedantic` isn't supported, but adding the --ghc-options manually doesn't work either, stack still prints the new flags as `[]`

Does this have something to do with `stack build --pedantic foo` resetting flags for other packages? Has this problem propagated to `stack ghci`? Running `stack ghci` and loading all packages is not an option for me unfortunately, a lot of modules makes it very slow.


Am I correct thinking this is what's going on? Is this expected behavior now? Any workarounds?

**Comment by chrisdone** [comment]
Right, `stack ghci` doesn't pass its arguments through to the build process, they're only passed to `ghc --interactive`. This would probably lead to unwelcome recompiling too if you provide stuff for `ghc --interactive` that has no relation to building of dependencies. And `stack build` doesn't save any state about the options you provide for `ghc-options` so you have to pass `--pedantic` every time and so should do so when calling `stack ghci` too.

Perhaps we need to distinguish between build flags and `ghc --interactive` flags. Maybe e.g.

* `--ghc-options` => passed to `ghc --interactive`
* `--build-ghc-options` => passed into the build process
* `--pedantic` => a shortcut for: `--ghc-options "-Wall --ghc-options -Werror" --build-ghc-options "-Wall --ghc-options -Werror"`

Then your regular use-case will work:

    $ stack build --pedantic
    $ stack ghci --pedantic

But also you can specify custom things in the rarer case like:

    $ stack build --ghc-options -fno-warn-orphans
    $ stack ghci --build-ghc-options -fno-warn-orphans --ghc-options -fobject-code

**Comment by mgsloan** [comment]
Looks like this got covered by the `build-ghc-options` flag, and furthermore will be addressed by #1186
